### PR TITLE
[POAE7-1473] enable pr title check in github actions

### PR DESCRIPTION
**Pull request by sijialouintel**

### CODE REVIEW

**Comment by sijialouintel** [comment]
Please review it @Intel-bigdata/bdf-cider-all

**Comment by jikunshang** [comment]
I just enable repo's github action settings, can you try retrigger it by pushing empty commit?
btw, you can reference this [link ](https://github.com/oap-project/sql-ds-cache/blob/master/.github/workflows/dev_cron.yml)for PR title check.

**Comment by sijialouintel** [comment]
> I just enable repo's github action settings, can you try retrigger it by pushing empty commit? btw, you can reference this [link ](https://github.com/oap-project/sql-ds-cache/blob/master/.github/workflows/dev_cron.yml)for PR title check.

Hi, I think this pr should be merged into cider first. After that, when another pr raised to cider, title check action will be triggered.

**Review by jikunshang — APPROVED** [review]

**Comment by sijialouintel** [comment]
added code style check

**Review by winningsix — APPROVED** [review]

**Comment by winningsix** [comment]
@sijialouintel Thanks for this PR. This enabled pre-commit. Please follow up in other PRs:
1. For license check, it failed due to some license title has corp name ahead of ASF license. Let's add some reg exp to pass it.
/*
 * Copyright 2020 OmniSci, Inc.
 *
2. Code style failed. It has quite a lot inconsistence in code style. Any way we only check newly commited code style? Otherwise, it will extremely hard to tell where my new code conflicted with code style.

File: .github/workflows/code-style-checker.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: "PR Code Style Checker"
+on: [pull_request]
+
+jobs:
```
Comment (jikunshang) [inline .github/workflows/code-style-checker.yml:21]: merge these two yaml file and rename? pr title check, code style check, license check.

File: .github/config_files/.licenserc.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+header:
+  license:
+    spdx-id: Apache-2.0
+    copyright-owner: Apache Software Foundation
+
+  paths-ignore:
```
Comment (jikunshang) [inline .github/config_files/.licenserc.yaml:23]: can check c++ related files(.cpp/.cc/.h ...) only ?

File: .github/config_files/.licenserc.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - '.clang-format'
+    - '.gitignore'
+    - 'dist'
+    - 'licenses'
+    - '**/*.MD'
+    - 'LICENSE'
+    - 'NOTICE'
+
+  comment: on-failure
```
Comment (jikunshang) [inline .github/config_files/.licenserc.yaml:32]: empty line

File: .github/config_files/.licenserc.yaml (hunk 2 above)
Comment (winningsix) [reply .github/config_files/.licenserc.yaml:23]: Usually, besides binary, it's supposed to have license header. I am not sure whether it applied for all projects.

Comment (jikunshang) [reply .github/config_files/.licenserc.yaml:23]: https://www.apache.org/licenses/LICENSE-2.0#apply  seems a LICENSE file is enough.

**Comment by winningsix** [comment]
Hi folks, close this and move discussion on internally repo.

**Pull request closed without merging**